dataloader/load_regression_data_from_arbitrary_gp.py

```diff
@@ -72,38 +85,51 @@ def gen(batch_size,
         yield (context_x, context_y, target_x), target_y
 
 
-class RegressionDataGeneratorUniform(RegressionDataGeneratorBase):
-    """Class that uses load_regression_data to create datasets.
-    """
+class RegressionDataGeneratorArbitraryGP(RegressionDataGeneratorBase):
+    """Class that generates a batch of data for regression based on
+    the original Conditional Neural Processes paper."""
     def __init__(self,
                  iterations: int=250,
                  batch_size: int=32,
                  min_num_context: int=3,
                  max_num_context: int=10,
                  min_num_target: int=2,
+                 min_val_uniform: int=-2,
+                 max_val_uniform: int=2,
                  kernel_length_scale: float=0.4):
         super().__init__(iterations=iterations, batch_size=batch_size)
         
+        
         self.min_num_context = min_num_context
         self.max_num_context = max_num_context
         self.min_num_target = min_num_target
+        
+        assert min_val_uniform < max_val_uniform, "min_val_uniform must be smaller than max_val_uniform"
+        self.min_val_uniform = min_val_uniform
+        self.max_val_uniform = max_val_uniform
+        
         self.kernel_length_scale = kernel_length_scale
         
         self.train_ds, self.test_ds = self.load_regression_data()
 
     
     def get_gp_curve_generator_from_uniform(self, testing: bool=False) -> Callable:
```

Comment:
should this name be changed as well?

Reply:
Or good catch! I'll do it right away.

Reply:
Good catch! I'll change them to `min_x_val_uniform` and `max_x_val_uniform` to make it more clear.